iosxr_iso2vbox.py

```diff
@@ -95,43 +97,53 @@ def main(argv):
+    parser.add_argument('ISO_FILE',
+                        help='local ISO filename or remote URI ISO filename...')
+    parser.add_argument('-a', '--artifactory', nargs='?', metavar="'New box reason'",
+                        const='No reason for update specified',
+                        help='Upload box to Artifactory. You can optionally specify a reason for uploading this box.')
+    parser.add_argument('-o', '--create_ova', action='store_true',
+                        help='additionally use vboxmanage to export an OVA')
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help='turn on verbose messages')
+
+    args = parser.parse_args()
+
+    # Handle Input ISO (Local or URI)
+    if re.search(':/', args.ISO_FILE):
+        # URI Image
+        cmdstring = 'scp %s@%s .' % (getpass.getuser(), args.ISO_FILE)
+        print('==> Will attempt to scp the remote image to current working dir. You may be required to enter your password.')
+        print('==> %s\n' % cmdstring)
+        subprocess.call(cmdstring, shell=True)
+        input_iso = os.path.basename(args.ISO_FILE)
+    else:
+        # Local image
+        input_iso = args.ISO_FILE
 
-    # Suck in the input ISO and handle errors
-    try:
-        opts, args = getopt.getopt(argv, 'hva:oi:s:', ['help', 'verbose', 'artifactory=', 'ova', 'iso=', 'scp='])
-    except getopt.GetoptError:
-        print(master_opts)
-        sys.exit(2)
-    for opt, arg in opts:
-        if opt in ('-h', '--help'):
-            print('==> A tool to create a VirtualBox box from an ISO')
-            print(master_opts)
-            print('==> E.g. iosxr-xrv64-vbox/iosxr_iso2vbox.py -iso iosxrv-fullk9-x64.iso')
-            print('==> E.g. iosxr-xrv64-vbox/iosxr_iso2vbox.py -scp server:<path>/iosxrv-fullk9-x64.iso')
-            sys.exit()
-        elif opt in ('-i', '--iso'):
-            input_iso = arg
-        elif opt in ('-s', '--scp'):
-            iso_to_scp = arg
-            cmdstring = 'scp %s@%s .' % (getpass.getuser(), iso_to_scp)
-            print('==> Will attempt to scp the remote image to current working dir. You may be required to enter your password.')
-            time.sleep(2)
-            print('==> %s\n' % cmdstring)
-            subprocess.call(cmdstring, shell=True)
-            input_iso = os.path.basename(iso_to_scp)
-        elif opt in ('-a', '--artifactory'):
-            copy_to_artifactory = True
-            reason = arg
-            if not reason:
-                reason = 'No reason for update specified'
-        elif opt in ('-v', '--verbose'):
-            verbose = True
-        elif opt in ('-o', '--ova'):
-            create_ova = True
-        else:
-            print(master_opts)
-            sys.exit()
+    # Handle artifactory
+    if args.artifactory is not None:
+        copy_to_artifactory = True
+        if args.artifactory is not 'No reason for update specified':
+            artifactory_reason = args.artifactory
```

Comment:
else, artifactory_reason == ''? Is that correct or should artifactory_reason be 'No reason for update specified'?